client/state/themes/actions.js

```diff
@@ -194,39 +195,79 @@ export function receiveThemes( data, site, queryParams, responseTime ) {
 	};
 }
 
-export function activate( theme, site, source = 'unknown' ) {
+/**
+ * Returns an action object to be used in signalling that a theme activation
+ * has been triggered
+ *
+ * @param  {String}  themeId Theme to be activated
+ * @param  {Object}  siteId Site used for activation
+ * @return {Object}  Action object
+ */
+export function themeActivation( themeId, siteId ) {
+	return {
+		type: THEME_ACTIVATE_REQUEST,
+		themeId,
+		siteId,
+	};
+}
+
+/**
+ * Returns an action object to be used in signalling that a theme activation
+ * has been successfull
+ *
+ * @param  {Object}  theme Theme received
+ * @param  {Number}  siteId Site used for activation
+ * @return {Object}  Action object
+ */
+export function themeActivated( theme, siteId ) {
```

Comment:
I am OK with that - just a bit sad since it reverts basically all the changes ( cleanup in my opinion ) I did :crying_cat_face: